### PR TITLE
SAK-46921 Sections TAs should only be able to assign students to the section they are a member of

### PR DESCRIPTION
**Pull request by alejandromf**
https://sakaiproject.atlassian.net/browse/SAK-46921

### CODE REVIEW

File: sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/AddSectionsBean.java [hunk 1]
```diff
@@ -496,4 +505,17 @@ protected void initDaysOfWeek(){
        DateFormatSymbols dfs = DateFormatSymbols.getInstance(rl.getLocale());
        daysOfWeek = dfs.getWeekdays();
     }
+
+	static void initReadOnlyCategories(ServerConfigurationService serverConfigurationService) {
+		if (readOnlyCategories == null) {
+			readOnlyCategories = new HashSet<>();
+			if (serverConfigurationService != null) {
+				String[] readOnlySectionCategories = serverConfigurationService
+						.getStrings(READ_ONLY_SECTION_CATEGORIES);
```
Comment (bjones86) [inline sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/AddSectionsBean.java:514]: Please document this new property in default.sakai.properties

Comment (SedueRey) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/AddSectionsBean.java:514]: Hello, @bjones86 , I've been talking to @alejandromf about this issue as we said yesterday and it's not a new property, `section.info.readonly.section.categories`  is already defined in default.sakai.properties, you can watch it here in https://github.com/sakaiproject/sakai/blob/master/config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties#L3455

It's a permission problem, the TA rol has more permission that it should, if it's assigned to a group he/she could modify groups they no belong to (this does not happen in resources, announcements or assignments).

Thanks

Comment (bjones86) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/AddSectionsBean.java:514]: Thanks, @SedueRey

**Review by juanjmerono — APPROVED** [review]

File: sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditStudentSectionsBean.java [hunk 2]
```diff
@@ -138,10 +143,11 @@ public void init() {
...
 			boolean member = isEnrolledInSection(enrolled, section);
 			boolean memberOtherSection = isEnrolledInOtherSection(enrolled, section);
-			
-			StudentSectionDecorator decoratedSection = new StudentSectionDecorator(
-					section, catName, taNames, totalEnrollments, member, memberOtherSection, showNegativeSpots);
-			
+
+			StudentSectionDecorator decoratedSection = new StudentSectionDecorator(section, catName, taNames,
+					totalEnrollments, member, memberOtherSection, showNegativeSpots,
+					(taUids.contains(getUserUid()) || (intructorUids.contains(getUserUid()))));
```
Comment (ottenhoff) [inline sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditStudentSectionsBean.java:149]: What if we are the superuser (admin)? 

CC: @ern

Comment (alejandromf) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditStudentSectionsBean.java:149]: You are right. If you are superadmin now you can't manage groups. I have created a SAK to fix this

https://sakaiproject.atlassian.net/browse/SAK-47854

Comment (alejandromf) [reply sections/sections-app/src/java/org/sakaiproject/tool/section/jsf/backingbean/EditStudentSectionsBean.java:149]: Meanwhile you can use the "Become user" tool to login as an Instructor

**Pull request merged**